### PR TITLE
Fixes #946 try catch follow up

### LINKED ISSUE #946: Encoding issue: trycatch aroung getUnitEnums so that loading works also on chinese/japanese OS

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: R/utilities-units.R [hunk 1]
```diff
@@ -288,6 +288,7 @@ getUnitsForDimension <- function(dimension) {
   validateIsString(dimension)
   dimensionTask <- getDimensionTask()
   rClr::clrCall(dimensionTask, "AllAvailableUnitNamesFor", enc2utf8(dimension))
+  stop("errror")
```
Comment (msevestre) [inline R/utilities-units.R:291]: @IndrajeetPatil THis is on purpose just to ask you why the other code does not work

File: R/utilities-units.R [hunk 2]
```diff
@@ -347,14 +348,18 @@ getUnitsEnum <- function() {
...
-        warning(messages$errorLoadingUnitsForDimension(dimension, cond))
+        errors <- c(errors, dimension)
         # making sure that in this case, the user sees that something went wrong
         return(c("Unavailable"))
       }
     )
     return(enum(replace(x, x == "", "Unitless")))
   })
 
+  if (length(errors) > 0L) {
```
Comment (msevestre) [inline R/utilities-units.R:359]: @IndrajeetPatil The errors vector is always empty. I am a bit confused here as I thought R would work with closure. What  am I doing wrong here. In this logic, I am collecting all errors at the function level and showing only one message. But it does not work.

File: R/utilities-units.R [hunk 3]
```diff
@@ -347,14 +348,18 @@ getUnitsEnum <- function() {
...
+    message(messages$errorLoadingUnitsForDimensionoadingUnitsForDimension(errors))
```
Comment (IndrajeetPatil) [inline R/utilities-units.R:360]:
```suggestion
    message(messages$errorLoadingUnitsForDimension(errors))
```

File: R/utilities-units.R (hunk 2 above)
Comment (IndrajeetPatil) [reply R/utilities-units.R:359]: Hmm, at face value, this seems perfectly fine. I will see if I can figure out the issue locally.

File: R/utilities-units.R (hunk 3 above)
Comment (msevestre) [reply R/utilities-units.R:360]: sure this is copy paste BS. But why isn't the code working?

File: R/utilities-units.R [hunk 4]
```diff
@@ -347,14 +348,18 @@ getUnitsEnum <- function() {
         getUnitsForDimension(dimension = dimension)
       },
       error = function(cond) {
-        warning(messages$errorLoadingUnitsForDimension(dimension, cond))
+        errors <- c(errors, dimension)
```
Comment (IndrajeetPatil) [inline R/utilities-units.R:351]: You need to do super-assignment here:

```suggestion
        errors <<- c(errors, dimension)
```

**Comment by msevestre** [comment]
What is super assignment?

**Comment by IndrajeetPatil** [comment]
> What is super assignment?

Ref: https://adv-r.hadley.nz/environments.html

image

**Comment by msevestre** [comment]
Ok cool

**Pull request merged**